### ISSUE TITLE
OCaml 5.2 support

### ISSUE THREAD

**Issue by kit-ty-kate** [OCamlPro/ocp-index#168]
The latest release of ocp-index currently fails with the following error when using OCaml 5.2:
```
#=== ERROR while compiling ocp-index.1.3.5 ====================================#
# context              2.2.0~beta2~dev | linux/x86_64 | ocaml-variants.5.2.0+trunk | file:///home/opam/opam-repository
# path                 ~/.opam/5.2/.opam-switch/build/ocp-index.1.3.5
# command              ~/.opam/5.2/bin/dune build -p ocp-index -j 1
# exit-code            1
# env-file             ~/.opam/log/ocp-index-19-80ca4b.env
# output-file          ~/.opam/log/ocp-index-19-80ca4b.out
### output ###
# (cd _build/default && /home/opam/.opam/5.2/bin/ocamlc.opt -w -40 -w -9 -g -bin-annot -I libs/.indexLib.objs/byte -I /home/opam/.opam/5.2/lib/bytes -I /home/opam/.opam/5.2/lib/ocaml/compiler-libs -I /home/opam/.opam/5.2/lib/ocp-indent/lexer -I /home/opam/.opam/5.2/lib/ocp-indent/utils -intf-suffix .ml -no-alias-deps -o libs/.indexLib.objs/byte/indexBuild.cmo -c -impl libs/indexBuild.pp.ml)
# File "libs/indexBuild.ml", line 450, characters 4-23:
# Error: The constructor "Types.Type_abstract" expects 1 argument(s),
#        but is applied here to 0 argument(s)
# (cd _build/default && /home/opam/.opam/5.2/bin/ocamlc.opt -w -40 -w -9 -g -bin-annot -I libs/.indexLib.objs/byte -I /home/opam/.opam/5.2/lib/bytes -I /home/opam/.opam/5.2/lib/ocaml/compiler-libs -I /home/opam/.opam/5.2/lib/ocp-indent/lexer -I /home/opam/.opam/5.2/lib/ocp-indent/utils -intf-suffix .ml -no-alias-deps -o libs/.indexLib.objs/byte/indexPredefined.cmo -c -impl libs/indexPredefined.pp.ml)
# File "libs/indexPredefined.ml", line 35, characters 24-89:
# Error: This expression has type
#          "('a * (Asttypes.variance * Asttypes.injectivity)) list"
#        but an expression was expected of type "Outcometree.out_type_param list"
#        Type "'a * (Asttypes.variance * Asttypes.injectivity)"
#        is not compatible with type "Outcometree.out_type_param"
# (cd _build/default && /home/opam/.opam/5.2/bin/ocamlopt.opt -w -40 -w -9 -g -I libs/.indexLib.objs/byte -I libs/.indexLib.objs/native -I /home/opam/.opam/5.2/lib/bytes -I /home/opam/.opam/5.2/lib/ocaml/compiler-libs -I /home/opam/.opam/5.2/lib/ocp-indent/lexer -I /home/opam/.opam/5.2/lib/ocp-indent/utils -intf-suffix .ml -no-alias-deps -o libs/.indexLib.objs/native/indexPredefined.cmx -c -impl libs/indexPredefined.pp.ml)
# File "libs/indexPredefined.ml", line 35, characters 24-89:
# Error: This expression has type
#          "('a * (Asttypes.variance * Asttypes.injectivity)) list"
#        but an expression was expected of type "Outcometree.out_type_param list"
#        Type "'a * (Asttypes.variance * Asttypes.injectivity)"
#        is not compatible with type "Outcometree.out_type_param"
```

**Comment by kit-ty-kate** [comment]
This was apparently fixed by https://github.com/OCamlPro/ocp-index/pull/169. Could we have a release? I can handle the opam-repository side of the work whenever a tag is accessible if that helps

**Comment by kit-ty-kate** [comment]
Fixed and released in 1.3.6
